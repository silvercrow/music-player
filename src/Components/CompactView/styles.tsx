import styled from 'styled-components';

interface Props {
  visible: boolean;
}

export const CompactBox = styled.div<Props>`
  display: ${(p) => (p.visible === true ? 'block' : 'none')};
  position: relative;
  width: 100%;
  height: 76px;
  bottom: 0;
  background-color: #3c4553;
  padding: 16px 20px;
  margin-top: auto;
  background-color: #5b6467;
  background-image: linear-gradient(315deg, #5b6467 0%, #8b939a 74%);
  border-radius: 10px;
`;
export const ExtendButton = styled.button`
  position: absolute;
  left: 42%;
  top: -20px;
  padding: 12px 0;
  border-radius: 50%;

  background-color: #5b6467;

  transition: all 0.3s ease;
  &:hover {
    top: -30px;
    transition: all 0.3s ease;
  }
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-size: 20px;
  background-position: center;
  background-color: #ffcd00;
`;
export const Arrow = styled.span`
  position: absolute;
  left: 12%;
  top: 5px;
  height: 30px;
  width: 30px;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='%23fff' strokeLinecap='round' strokeLinejoin='round' strokeWidth='2' class='feather feather-chevron-up' width='' viewBox='0 0 24 24'%3E%3Cdefs/%3E%3Cpath d='M18 15l-6-6-6 6'/%3E%3C/svg%3E");
`;

export const Content = styled.div`
  display: flex;
  align-items: center;
`;
export const ArtCover = styled.div`
  img {
    border-radius: 10px;
    width: 32px;
    height: 32px;
    -o-object-fit: cover;
    object-fit: cover;
    margin-right: 16px;
  }
`;
export const InfoWrapper = styled.div``;
export const Title = styled.div`
  font-size: 12px;
  line-height: 18px;
  font-weight: 800;
  color: #fff;
`;
export const Subtitle = styled.div`
  font-size: 10px;
  color: #fff;
`;
export const ButtonsWrapper = styled.div`
  margin-left: auto;
  display: flex;
`;
export const StopButton = styled.button`
  background-color: #ffff;
  height: 15px;
  width: 15px;
  cursor: pointer;
  margin-right: 24px;
`;
export const NextButton = styled.div`
  color: #fff;
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
`;
export const Progress = styled.div`
  display: flex;
  align-items: center;
  padding: 6px 0;
`;
export const CurrentTime = styled.div`
  display: inline-block;
  font-size: 8px;
  line-height: 12px;
  color: #fff;
`;
export const ProgressBar = styled.div`
  height: 2px;
  width: 100%;
  margin: 0 8px;
  background-color: #1f2128;
  border-radius: 2px;
  overflow: hidden;
`;
export const ProgressFill = styled.div`
  display: block;
  position: relative;
  width: 40%;
  height: 100%;
  background-color: #ffcd00;
  border-radius: 6px;
`;
export const TimeLeft = styled.div`
  display: inline-block;
  font-size: 8px;
  line-height: 12px;
  color: #fff;
`;
