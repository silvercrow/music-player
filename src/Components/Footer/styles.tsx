// src/components/Navbar/styles.tsx
import styled from 'styled-components';

export const Container = styled.div`
  position: fixed;
  bottom: 0;
  height: 70px;
  color: #f1f1f1;
  width: 100%;
  background: #46505f;
`;

export const FooterNav = styled.div`
  display: flex;
  justify-content: flex-end;
`;

export const FooterNavItem = styled.div`
  margin: 30px 30px 0px 0px;
  font-family: Merriweather;
  font-style: normal;
  font-weight: bold;
  font-size: 14px;
  display: flex;
  align-items: center;
  text-align: center;
  text-transform: uppercase;
  letter-spacing: 0.2em;
`;
