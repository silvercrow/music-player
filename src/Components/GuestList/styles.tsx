import styled from 'styled-components';

export const GuestlistBox = styled.ul`
  padding: 0;
  list-style-type: none;
  margin-top: 20px;
  height: 450px;
  overflow-y: scroll;

  &::-webkit-scrollbar {
    -webkit-appearance: none;
    width: 10px;
  }

  &::-webkit-scrollbar-thumb {
    border-radius: 5px;
    background-color: rgba(0, 0, 0, 0.5);
    -webkit-box-shadow: 0 0 1px rgba(255, 255, 255, 0.5);
  }
`;

export const Table = styled.table`
  border-spacing: 1;
  border-collapse: collapse;
  background: white;
  border-radius: 10px;
  overflow: hidden;
  width: 100%;
  margin: 0 auto;
  margin-top: 10px;
  position: relative;
`;
export const Thead = styled.thead`
  font-family: OpenSans-Regular;
  font-size: 18px;
  color: #000;
  line-height: 1.2;
  font-weight: unset;
`;
export const TR = styled.tr`
  height: 60px;
  background-color: #ffff;
  height: 50px;
  &:hover {
    -webkit-box-shadow: inset 1px 1px 52px -24px #ababab, 5px 5px 5px 1px #dddddd;
    -moz-box-shadow: inset 1px 1px 52px -24px #ababab, 5px 5px 5px 1px #dddddd;
    -o-box-shadow: inset 1px 1px 52px -24px #ababab, 5px 5px 5px 1px #dddddd;
    background-color: #bdcad9;
    background-image: linear-gradient(315deg, #bdcad9 0%, #e1dada 74%);
    
`;
export const TH = styled.th`
  padding-left: 8px;
  text-align: left;
`;
export const Tbody = styled.tbody``;
export const TD = styled.td`
  padding-left: 8px;
  text-align: left;
  border-bottom: solid 1px #bdcad9;
`;
