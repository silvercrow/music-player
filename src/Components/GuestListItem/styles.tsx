import styled from 'styled-components';

export const ListItem = styled.div`
  display: flex;
  align-items: center;
  margin: 0;
  padding: 10px 20px;
  border-top: 1px solid #fff;
  transition: all 0.2s ease-in;
  overflow: auto;
  background-color: #f6f6f6;
  background-image: linear-gradient(315deg, #f6f6f6 0%, #e9e9e9 74%);

  &:hover {
    box-shadow: inset 1px 1px 52px -24px #ababab, 5px 5px 5px 1px #dddddd;
    -webkit-box-shadow: inset 1px 1px 52px -24px #ababab, 5px 5px 5px 1px #dddddd;
    -moz-box-shadow: inset 1px 1px 52px -24px #ababab, 5px 5px 5px 1px #dddddd;
    -o-box-shadow: inset 1px 1px 52px -24px #ababab, 5px 5px 5px 1px #dddddd;
    background-color: #d9d9d9;
    background-image: linear-gradient(315deg, #d9d9d9 0%, #f6f2f2 74%);
  }
`;

export const Info = styled.span`
  margin-left: 30px;
`;
