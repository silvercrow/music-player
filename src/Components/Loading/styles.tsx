import styled from 'styled-components';

export const Page = styled.div`
  width: 100%;
  margin: 0 auto;
`;

export const Container = styled.div`
  width: 100%;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  padding: 15px;
  background-image: linear-gradient(315deg, #f6f6f6 0%, #e9e9e9 74%);
`;

export const Wrapper = styled.div`
  background: #fff;
  border-radius: 10px;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 30px;
`;

export const Loader = styled.div`
  font-size: 10px;
`;
