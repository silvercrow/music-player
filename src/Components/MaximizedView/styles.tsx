import styled from 'styled-components';

interface MaximazedProps {
  visible: boolean;
}

export const MaximizedBox = styled.div<MaximazedProps>`
  position: absolute;
  top: 0;
  z-index: 2;
  height: 100%;
  width: 100%;
  padding: 20px;
  transition: all 0.4s ease;
  top: 0;
  transform: ${(p) => (p.visible === true ? 'translatey(0)' : 'translatey(100%)')};
  transition: all 0.4s ease;
  background-color: #bdcad9;
  background-image: linear-gradient(315deg, #bdcad9 0%, #e1dada 74%);
  border: solid 1px white;
  border-radius: 30px;
`;
export const CloseButtonBox = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

export const CloseButton = styled.button`
  display: flex;
  align-items: center;
  justify-content: center;
  border-radius: 50%;
  transition: all 0.3s ease;
  &:hover {
    background-color: #ffcd00;
    transition: all 0.3s ease;
  }
  width: 40px;
  height: 40px;
  background-color: #ffff;
`;

export const Arrow = styled.span`
  position: relative;
  height: 30px;
  width: 30px;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' fill='none' stroke='%23536383' strokeLinecap='round' strokeLinejoin='round' strokeWidth='2' class='feather feather-chevron-down' viewBox='0 0 24 24'%3E%3Cdefs/%3E%3Cpath d='M6 9l6 6 6-6'/%3E%3C/svg%3E");
`;
export const Title = styled.div`
  display: block;
  color: #fff;
  font-size: 14px;
  font-weight: 600;
  text-align: center;
  margin-bottom: 10px;
  text-shadow: 0px 0px 7px rgba(150, 150, 150, 1);
`;
export const Cover = styled.div`
  margin-top: 15px;
  position: relative;
  border-radius: 20px;
  width: 100%;
  display: flex;
  justify-content: center;
`;
export const Shadow = styled.div`
  img {
    position: absolute;
    top: 0;
    left: 50%;
    transform: translateX(-46%);
    filter: blur(50px);
    width: 100%;
    height: 190px;
    z-index: -1;
  }
`;
export const Art = styled.div`
  img {
    margin: 0 auto;
    -o-object-fit: cover;
    object-fit: cover;
    width: 200px;
    height: 270px;
    border: 0;
    border-radius: 20px;
    position: relative;
    z-index: 2;
    box-shadow: 0px 0px 3px #000;
    -moz-box-shadow: 0px 0px 3px #000;
    -webkit-box-shadow: 0px 0px 3px #000;
    -khtml-box-shadow: 0px 0px 3px #000;
  }
`;
export const SongInfo = styled.div`
  text-align: center;
`;
export const SongTitle = styled.div`
  margin-top: 10px;
  color: #000;
  font-weight: 600;
`;
export const SongSubTitle = styled.div`
  color: #59657c;
  font-size: 12px;
`;
export const MaxProgress = styled.div`
  margin: 0 auto;
  padding-top: 30px;
  max-width: 80%;
  display: flex;
  align-items: center;
  padding: 6px 0;
`;
export const CurrentTime = styled.span`
  display: inline-block;
  font-size: 8px;
  line-height: 12px;
  color: #fff;
  color: #59657c;
`;
export const ProgressBar = styled.div`
  height: 2px;
  width: 100%;
  margin: 0 8px;
  background-color: #1f2128;
  border-radius: 2px;
  overflow: hidden;
  border-radius: 6px;
  height: 4px;
  background-color: #323542;
`;
export const ProgressFill = styled.div`
  display: block;
  position: relative;
  width: 40%;
  height: 100%;
  background-color: #ffcd00;
  border-radius: 6px;
`;
export const TimeLeft = styled.span`
  display: inline-block;
  font-size: 8px;
  line-height: 12px;
  color: #fff;
  color: #59657c;
`;
export const ButtonsBar = styled.div`
  display: flex;
  align-items: center;
  justify-content: space-between;
  margin-top: 24px;
`;
export const PreviousSong = styled.button`
  width: 24px;
  height: 20px;
  color: #fff;
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24' fill='%23424665' stroke='%23424665' strokeLinecap='round' strokeLinejoin='round' strokeWidth='2' class='feather feather-skip-back'%3E%3Cdefs/%3E%3Cpath d='M19 20L9 12l10-8v16zM5 19V5'/%3E%3C/svg%3E");
`;

interface PlayPauseProps {
  isPlaying: boolean;
}

export const PlayButton = styled.button<PlayPauseProps>`
  display: ${(p) => (p.isPlaying === true ? 'none' : 'block')};
  width: 40px;
  height: 40px;
  color: #fff;
  margin: 0 auto;
  background-color: #ffcd00;
  border-radius: 12px;
  padding: 4px;

  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='20' height='20' viewBox='0 0 24 24' stroke='%23fff' strokeWidth='2' strokeLinecap='round' strokeLinejoin='round' class='feather feather-play' fill='%23fff'%3E%3Cpath d='M5 3l14 9-14 9V3z'%3E%3C/path%3E%3C/svg%3E");
  background-repeat: no-repeat;
  background-position: center;
`;

export const StopButton = styled.button<PlayPauseProps>`
  display: ${(p) => (p.isPlaying === true ? 'block' : 'none')};
  width: 40px;
  height: 40px;
  color: #fff;
  margin: 0 auto;
  background-color: #ffcd00;
  border-radius: 12px;
  padding: 4px;

  background-repeat: no-repeat;
  background-position: center;
`;

export const Square = styled.div`
  margin: 0 auto;
  background-color: #ffff;
  height: 14px;
  width: 14px;
  cursor: pointer;
`;

export const NextSong = styled.button`
  width: 24px;
  height: 20px;
  color: #fff;
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  background-image: url("data:image/svg+xml,%3Csvg xmlns='http://www.w3.org/2000/svg' width='24' height='24' fill='%23424665' stroke='%23424665' strokeLinecap='round' strokeLinejoin='round' strokeWidth='2' class='feather feather-skip-back'%3E%3Cdefs/%3E%3Cpath d='M19 20L9 12l10-8v16zM5 19V5'/%3E%3C/svg%3E");
  transform: rotate(180deg);
`;
export const VolumeBars = styled.div`
  display: flex;
  align-items: center;
  margin-top: 24px;
`;
export const Volume = styled.div`
  height: 2px;
  width: 100%;
  margin: 0 8px;
  background-color: #1f2128;
  border-radius: 2px;
  overflow: hidden;
  overflow: visible;
  cursor: pointer;
`;
export const VolumeTitle = styled.div`
  color: #59657c;
  font-size: 12px;
`;
export const PercentageFill = styled.div`
  display: block;
  position: relative;
  width: 40%;
  height: 100%;
  background-color: #6343e4;
  border-radius: 6px;

  &:after {
    content: '';
    position: absolute;
    width: 16px;
    height: 16px;
    border-radius: 50%;
    background-color: #6343e4;
    right: 0;
    top: 50%;
    transform: translatey(-50%);
  }
`;

export const RangeBar = styled.div`
  margin-top: 20px;
  border: solid 1px #bdcad9;
  border-radius: 10px;
  padding: 10px;
  background-color: #bdcad9;
  background-image: linear-gradient(315deg, #bdcad9 0%, #e1dada 64%);
`;
