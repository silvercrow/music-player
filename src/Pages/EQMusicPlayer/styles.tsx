import styled from 'styled-components';

export const MusicBox = styled.div`
  font-family: 'DM-sans', sans-serif;
  display: flex;
  justify-content: center;
  min-height: 80vh;
  margin-top: 20px;
`;

export const MusicPlayer = styled.div`
  position: relative;
  width: 320px;
  height: 620px;
  border-radius: 30px;
  box-shadow: 0px 0px 7px 1px rgba(40, 42, 13, 0.12);
  overflow: hidden;
  display: flex;
  flex-direction: column;
  background-color: #f6f6f6;
  background-image: linear-gradient(315deg, #f6f6f6 0%, #e9e9e9 74%);
  border: solid 1px #f6f6f6;
`;

export const Queue = styled.div`
  flex: 1;
  overflow: auto;
  height: calc(100% - 32px);
`;

export const PlayTitle = styled.div`
  padding: 0px 20px 6px 20px;
  display: flex;
  color: #626262;
  font-size: 18px;
  font-weight: bold;
  padding: 30px 20px 0 20px;
  text-shadow: 0px 0px 3px rgba(255, 255, 255, 1);
`;
