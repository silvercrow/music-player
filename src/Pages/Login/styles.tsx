import styled from 'styled-components';

export const Page = styled.div`
  width: 100%;
  margin: 0 auto;
`;

export const Container = styled.div`
  width: 100%;
  min-height: 100vh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  align-items: center;
  padding: 15px;
  background: #9053c7;
  background-color: #f39f86;
  background-image: linear-gradient(315deg, #f39f86 0%, #f9d976 74%);
`;

export const Wrapper = styled.div`
  background: #fff;
  border-radius: 10px;
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  display: flex;
  flex-wrap: wrap;
  justify-content: space-between;
  padding: 30px 45px 33px 60px;
`;

export const LoginBox = styled.div``;

export const Title = styled.span`
  font-size: 24px;
  color: #333333;
  line-height: 1.2;
  text-align: center;
  width: 100%;
  display: block;
  padding-bottom: 12px;
`;

export const LoginButtonContainer = styled.div`
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  display: flex;
  flex-wrap: wrap;
  justify-content: center;
  padding-top: 20px;
`;

export const LoginButton = styled.button`
  font-family: Montserrat-Bold;
  font-size: 15px;
  line-height: 1.5;
  color: #fff;
  text-transform: uppercase;
  width: 100%;
  height: 50px;
  border-radius: 25px;
  background: #57b846;
  display: -webkit-box;
  display: -webkit-flex;
  display: -moz-box;
  display: -ms-flexbox;
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 25px;
  -webkit-transition: all 0.4s;
  -o-transition: all 0.4s;
  -moz-transition: all 0.4s;
  transition: all 0.4s;
`;

export const GoogleButton = styled.div`
  display: inline-block;
  background: white;
  color: #444;
  border-radius: 5px;
  border: thin solid #888;
  box-shadow: 1px 1px 1px grey;
  white-space: nowrap;
  &:hover {
    cursor: pointer;
  }
`;

export const Label = styled.span`
  font-family: serif;
  font-weight: normal;
`;

export const Icon = styled.span`
  background: url('https://upload.wikimedia.org/wikipedia/commons/thumb/5/53/Google_%22G%22_Logo.svg/512px-Google_%22G%22_Logo.svg.png')
    transparent 5px 50% no-repeat;
  display: inline-block;
  vertical-align: middle;
  width: 34px;
  height: 34px;
  background-size: cover;
  padding: 7px;
`;

export const ButtonText = styled.span`
  display: inline-block;
  vertical-align: middle;
  padding-left: 42px;
  padding-right: 42px;
  font-size: 14px;
  font-weight: bold;
  /* Use the Roboto font that is loaded in the <head> */
  font-family: 'Roboto', sans-serif;
`;
