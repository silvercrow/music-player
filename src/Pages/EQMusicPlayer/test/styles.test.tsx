import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { MusicBox, MusicPlayer, Queue, PlayTitle } from '../styles';

test('MusicBox', () => {
  const tree = renderer.create(<MusicBox />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('MusicPlayer', () => {
  const tree = renderer.create(<MusicPlayer />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('Queue', () => {
  const tree = renderer.create(<Queue />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('PlayTitle', () => {
  const tree = renderer.create(<PlayTitle />).toJSON();
  expect(tree).toMatchSnapshot();
});
