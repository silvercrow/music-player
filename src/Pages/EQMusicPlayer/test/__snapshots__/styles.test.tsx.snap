// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MusicBox 1`] = `
.c0 {
  font-family: 'DM-sans',sans-serif;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 101vh;
  margin-top: 20px;
}

<div
  className="c0"
/>
`;

exports[`MusicPlayer 1`] = `
.c0 {
  position: relative;
  width: 320px;
  height: 620px;
  border-radius: 30px;
  box-shadow: 0px 0px 7px 1px rgba(40,42,13,0.12);
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background-color: #f6f6f6;
  background-image: linear-gradient(315deg,#f6f6f6 0%,#e9e9e9 74%);
  border: solid 1px #f6f6f6;
}

<div
  className="c0"
/>
`;

exports[`PlayTitle 1`] = `
.c0 {
  padding: 0px 20px 6px 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  color: #626262;
  font-size: 18px;
  font-weight: bold;
  padding: 30px 20px 0 20px;
  text-shadow: 0px 0px 3px rgba(255,255,255,1);
}

<div
  className="c0"
/>
`;

exports[`Queue 1`] = `
.c0 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  overflow: auto;
  height: calc(100% - 32px);
}

<div
  className="c0"
/>
`;
